### PR TITLE
docs: move xds protocol to rst

### LINKED ISSUE #6338: Move XDS_PROTOCOL.md into sphinx docs
https://github.com/envoyproxy/envoy/blob/master/api/XDS_PROTOCOL.md

We should port this into the sphinx docs. It will be more discoverable, allow direct ref linking, etc.

cc @htuch @ramaraochavali

### PR DESCRIPTION
**Pull request by ramaraochavali**
Description: This PR moves the xds protocol from md to rst.
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Fixes #6338

### CODE REVIEW

File: docs/root/intro/intro.rst [hunk 1]
```diff
@@ -13,3 +13,4 @@ Introduction
   getting_help
   version_history
   deprecated
+
```
Comment (ramaraochavali) [inline docs/root/intro/intro.rst]: added this new line to be consistent with all other files

**Comment by ramaraochavali** [comment]
@mattklein123 @htuch  PTAL

File: docs/root/index.rst [hunk 2]
```diff
@@ -19,4 +19,5 @@ Envoy documentation
   operations/operations
   extending/extending
   api-v2/api
+  api-docs/xds_protocol
```
Comment (htuch) [inline docs/root/index.rst]: I think we should have a single top-level "API", and nested under this the xDS wire and API reference entries.

File: api/xds_protocol.rst [hunk 3]
```diff
@@ -0,0 +1,456 @@
+xDS REST and gRPC protocol
+==========================
+
+Envoy discovers its various dynamic resources via the filesystem or by
+querying one or more management servers. Collectively, these discovery
+services and their corresponding APIs are referred to as *xDS*.
+Resources are requested via *subscriptions*, by specifying a filesystem
+path to watch, initiating gRPC streams or polling a REST-JSON URL. The
+latter two methods involve sending requests with a :ref:`Discovery Request <envoy_api_msg_DiscoveryRequest>`
```
Comment (htuch) [inline api/xds_protocol.rst]: Why did you add a space here between "Discovery" and "Request" (and "Response" below)? Ideally we keep the content exactly the same as the previous, no changes to things like naming etc.

File: api/xds_protocol.rst [hunk 4]
```diff
@@ -0,0 +1,456 @@
...
+and the nonce provided by the management server. If the update was
+successfully applied, the :ref:`version_info <envoy_api_field_DiscoveryResponse.version_info>` will be **X**, as indicated
+in the sequence diagram:
+
+.. figure:: diagrams/simple-ack.svg 
+   :alt: Version update after ACK
+
+In this sequence diagram, and below, the following format is used to abbreviate messages: 
+
+- *DiscoveryRequest*: (V=`version_info`,R=`resource_names`,N=`response_nonce`,T=`type_url`)
```
Comment (htuch) [inline api/xds_protocol.rst]: The backticks are visible in the rendered docs here.

File: api/xds_protocol.rst [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+Later, an API update may succeed at a new version **Y**:
+
+
+.. figure:: diagrams/later-ack.svg
+   :alt: ACK after NACK
+
+Each stream has its own notion of versioning, there is no shared
+versioning across resource types. When ADS is not used, even each
+resource of a given resource type may have a distinct version, since the
+Envoy API allows distinct EDS/RDS resources to point at different :ref:`Config Sources <envoy_api_msg_core.ConfigSource>`.
```
Comment (htuch) [inline api/xds_protocol.rst]: This is another place a space has been inserted inside an identifier.

File: docs/root/index.rst (hunk 2 above)
Comment (ramaraochavali) [reply docs/root/index.rst]: done

File: api/xds_protocol.rst (hunk 3 above)
Comment (ramaraochavali) [reply api/xds_protocol.rst]: It was by mistake.  I removed the spaces now

**Comment by ramaraochavali** [comment]
@htuch  changed. PTAL.

**Comment by phylake** [comment]
All the links are broken now, and the whole thing is harder to read. I'm guessing those things are related

image

**Comment by moderation** [comment]
@phylake the finished product is rendered at https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol and the links look OK and work as expected.
